openhcl/host_fdt_parser/src/lib.rs

```diff
@@ -514,6 +517,16 @@ impl<
 
                                 storage.entropy = Some(entropy);
                             }
+                            // These parameters may not be present so it is not an error if they are missing.
+                            "servicing" => {
+                                storage.preserve_dma_4k_pages =
+                                    match openhcl_child.find_property("dma-preserve-pages") {
+                                        Ok(pages) => pages.map(|p| {
```

Comment:
what are the desired semantics here? namely, if "serivcing" is present, shouldn't "dma-preserve-pages" also be present? or are both the servicing field, and the underlying field both independently optional?

Reply:
> both the servicing field, and the underlying field both independently optional?

That's the idea.

Reply:
in that case, following up to the comment above - instead of using a nested `.map()`, you can use `and_then`.

(also, don't use `expect` to do error handling)

Reply:
I still couldn't figure out proper rust syntax. Main problem is that read_u64 returns Result<> so error path needs to be handled somehow...

Reply:
.transpose() is your friend

Reply:
@daprilik found the expression which works for this case. Unfortunately transpose() is different. Please take a look.

Reply:
Clippy gave a better suggestion to use .and_then() instead of .map() - now indeed looks better.